pkg/updater/slim_updater.go

```diff
@@ -0,0 +1,140 @@
+package updater
+
+import (
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/urlfmt"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/scanner/pkg/repo2cpe"
+	"github.com/stackrox/scanner/pkg/wellknowndirnames"
+)
+
+const (
+	repoToCPEFilename = "rhvel2/repository-to-cpe.json"
+)
+
+var (
+	slimUpdaterDir = filepath.Join(wellknowndirnames.WriteableDir, "slim-updater-artifacts.d")
+)
+
+// SlimUpdater updates the Scanner's vulnerability definitions, contacting
+// Sensor, instead of Central as the Updater does.
+type SlimUpdater struct {
+	interval        time.Duration
+	lastUpdatedTime time.Time
+	stopSig         *concurrency.Signal
+
+	sensorClient           *http.Client
+	repoToCPE              *repo2cpe.Mapping
+	repoToCPELocalFilename string
+	repoToCPEURL           string
+}
+
+// NewSlimUpdater creates and initialize a new slim updater.
+func NewSlimUpdater(updaterConfig Config, sensorEndpoint string, repoToCPE *repo2cpe.Mapping) (*SlimUpdater, error) {
+	// Get the most recent genesis dump UUID, and construct the update URL.
+	uuid, err := getMostRecentGenesisDumpUUID()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting genesis UUID")
+	}
+	repoToCPEURL, err := urlfmt.FullyQualifiedURL(
+		strings.Join([]string{
+			urlfmt.FormatURL(sensorEndpoint, urlfmt.HTTPS, urlfmt.NoTrailingSlash),
+			"/scanner/definitions",
+		}, "/"),
+		url.Values{
+			"uuid": []string{uuid},
+			"file": []string{repoToCPEFilename},
+		})
+	utils.CrashOnError(err)
+
+	// Create sensor's HTTP client.
+	//
+	// FIXME Adopt clientconn.NewHTTPClient() when available in the repository.
```

Comment:
Will be provided by https://github.com/stackrox/stackrox/pull/1420

Reply:
You can update the `go.mod` with the following:

1. In the `github.com/stackrox/rox => github.com/stackrox/stackrox` replace line, change the version to `github.com/stackrox/stackrox jvdm/ROX-9614/pull`
2. Run `go mod tidy`
3. You can use the `clientconn.NewHTTPClient()` form this point. You'd just have to be sure to update the version in the `go.mod` with `master` once your PR is merged and then `go mod tidy` again